runtime/compiler/env/PersistentAllocator.cpp

```diff
@@ -228,58 +324,256 @@ void
+               blockIterator->previous()->setNext(block);
+               blockIterator->setPrevious(block);
+               // Adjust the interval bounderies
+               if (_startInterval[index]->size() > block->size())
+                  {
+                  // block becomes the first entry in this interval
+                  TR_ASSERT(_startInterval[index] == blockIterator, "blockInterator must be first block in this interval"); 
+                  _startInterval[index] = block;
+                  }
+               else // Insert in the middle or end of interval
+                  {
+                  // blockIterator may belong to the next interval in
+                  // which case _endInterval needs to be changed to block
+                  if (_endInterval[index]->size() < block->size())
+                     {
+                     TR_ASSERT(getInterval(blockIterator->size()) > index, "blockInterator must be first block in the next interval: index=%zu b->sz=%zu bi->sz=%zu", index, block->size(), blockIterator->size());
+                     _endInterval[index] = block;
+                     }
+                  }
+               }
+            }
+         else // Insert after 'prev'; block will be the very last entry in the list
+            {
+            TR_ASSERT(prev, "prev must exist");
+            TR_ASSERT(!prev->next(), "this must be the last element in the list");
+            block->setPrevious(prev);
+            block->setNext(NULL);
+            prev->setNext(block);
+            _endInterval[index] = block;
+            }
+         }
+      else // There are no other blocks in this interval
+         {
+         // Find the closest interval that has a block
+         // I verified that the block will not be attached at the beginning of the list.
+         // Thus, there must be a previous interval with some block in it
+         TR_ASSERT(index > 0, "Index must be greater than 0");
+         for (int i = (int)index - 1; i >= 0; i--)
+            {
+            if (_endInterval[i])
+               {
+               block->setPrevious(_endInterval[i]);
+               block->setNext(_endInterval[i]->next());
+               TR_ASSERT(block->size() > _endInterval[i]->size(), "wrong sizes previous");
+               if (block->next())
+                  {
+                  block->next()->setPrevious(block);
+                  TR_ASSERT(block->size() < block->next()->size(), "wrong sizes after block->size()=%zu next->size()=%zu", block->size(), block->next()->size());
+                  }
+               _endInterval[i]->setNext(block);
+               break;
+               }
+            }
+         TR_ASSERT(block->previous(), "I must have attached my block to the list");
+         _startInterval[index] = block;
+         _endInterval[index] = block;
+         }
       }
+      checkIntegrity("freeVariableSizeBlock end");   
    }
 
+void 
+PersistentAllocator::checkIntegrity(const char msg[])
+   {
+#ifdef DEBUG
```

Comment:
Since we don't really build debug builds, maybe this is better guarded with an option, or env var. Otherwise, we risk this code getting stale.

Reply:
My initial intent was to delete this function altogether once testing was done.
Then, I reconsidered: if we ever suspect a bug in this code I may want this code reinstated and since it took me some time to write it, I don't want to spend that time again.
However, I also don't want any additional overhead since this allocation code is called very frequently for JITServer. I would like the compiler to not even call this function and the only I can do that is if the compiler sees an empty function that inlines to nothing. Adding an `if` statement based on an option will not achieve my goal.
I could change `DEBUG` into some other preprocessor variable that is not defined by default.